From c2be4c6c4eb92cb64405b01bcc25bc856f8b9a92 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@protonmail.com>
Date: Sun, 2 Mar 2025 14:02:58 -0700
Subject: [PATCH 1/1] Fix for mtl-2.3

Control.Monad, Control.Monad.IO.Class, and Control.Monad.Trans.Class
are no longer implicit exports of Control.Monad.* from mtl

Signed-off-by: hololeap <hololeap@protonmail.com>
--- a/demo/AudioSin.hs
+++ b/demo/AudioSin.hs
@@ -9,6 +9,7 @@ import           Codec.FFmpeg.Enums
 import           Codec.FFmpeg.Juicy
 import           Codec.FFmpeg.Types
 import           Codec.Picture
+import           Control.Monad
 import           Control.Monad.Except
 import           Data.IORef
 import           Foreign.C.Types
--- a/src/Codec/FFmpeg/Decode.hs
+++ b/src/Codec/FFmpeg/Decode.hs
@@ -13,6 +13,7 @@ import           Codec.FFmpeg.Scaler
 import           Codec.FFmpeg.Types
 import           Control.Arrow             (first)
 import           Control.Monad             (void, when)
+import           Control.Monad.IO.Class
 import           Control.Monad.Except
 import           Control.Monad.Trans.Maybe
 import           Foreign.C.String
--- a/src/Codec/FFmpeg/Juicy.hs
+++ b/src/Codec/FFmpeg/Juicy.hs
@@ -9,7 +9,9 @@ import Codec.FFmpeg.Enums
 import Codec.FFmpeg.Internal.Linear (V2(..))
 import Codec.FFmpeg.Types
 import Control.Arrow (first)
-import Control.Monad ((>=>))
+import Control.Monad ((>=>), guard)
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Class
 import Control.Monad.Except
 import Control.Monad.Trans.Maybe
 import Data.Foldable (traverse_)
--- a/src/Codec/FFmpeg/Probe.hsc
+++ b/src/Codec/FFmpeg/Probe.hsc
@@ -18,6 +18,7 @@ module Codec.FFmpeg.Probe (
 
 import Control.Applicative ( Applicative )
 import Control.Monad.Catch ( MonadMask, finally )
+import Control.Monad
 import Control.Monad.Reader
 import Control.Monad.Trans.Except
 import Data.Int ( Int64 )
-- 
2.45.3

